segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSStorage.java

```diff
@@ -408,7 +412,8 @@ public SegmentHandle openWrite(String streamSegmentName) throws StreamSegmentExc
             } catch (IOException e) {
                 throw HDFSExceptionHelpers.convertException(streamSegmentName, e);
             }
-        } while (fencedCount < MAX_ATTEMPT_COUNT);
+            // Looping for the maximum possible number.
+        } while (fencedCount < (int) (MAX_EPOCH - 1));
```

Comment:
Why are you casting Max_Long - 1 to an int?
(Also this value is so huge, this might as well be while(true)

Reply:
Fixed. Dropped the cast. checkstyle does not like while(true).